File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
def solution(numbers):
    a = map(lambda x: [str(x)*(12//len(str(x))),len(str(x))], numbers)
    b = sorted(a, reverse=True)
    answer = ''
    
    for i in b:
        answer += i[0][:i[1]]    
    return str(int(answer))
'''
from collections import deque
def solution(numbers):
    s = deque()
    answer = ''
    
    for i in numbers:
        if len(s) == 0:
            s.append(str(i))
            continue
        cnt = 0
        for j in s:
            if int(str(j)+str(i)) <= int(str(i)+str(j)):
                break
            else:
                cnt +=1
        s.insert(cnt, str(i))
    return ''.join(s)
'''
            
    
            

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.js
function solution(numbers) {
    var answer = '';
    
    const arr = numbers.map((e,i)=> {
        let s = e.toString()
        let ss = ""
        if(s.length === 1) {
            ss = s.repeat(12)
        } else if(s.length === 2) {
            ss = s.repeat(6)
        } else if(s.length === 3) {
            ss = s.repeat(4)
        } else {
            ss = s.repeat(3)
        }
        
        return [ss,i]
    })
    
    arr.sort((a,b)=> b[0]-a[0])
    
    arr.forEach(e=>{
        answer += numbers[e[1]].toString()
    })
    return answer[0] === "0" ? "0": answer
}